commit 76e5216e317f39da2bc5bf905721cd9554ee6d09
Author: Joseph Myers <joseph@codesourcery.com>
Date:   Mon Jun 23 15:48:42 2014 +0000

    Update headers for Linux 3.15.
    
    This patch updates glibc headers for changes / new definitions in
    Linux 3.15.  In the course of my review I noticed that
    IPV6_PMTUDISC_INTERFACE was absent from glibc despite the inclusion of
    IP_PMTUDISC_INTERFACE; I added it along with IP_PMTUDISC_OMIT and
    IPV6_PMTUDISC_OMIT.  I did not add FALLOC_FL_NO_HIDE_STALE given the
    kernel header comment that it is reserved.
    
    Tested x86_64.
    
            * sysdeps/unix/sysv/linux/bits/fcntl-linux.h [__USE_GNU]
            (FALLOC_FL_COLLAPSE_RANGE): New macro.
            [__USE_GNU] (FALLOC_FL_ZERO_RANGE): Likewise.
            * sysdeps/unix/sysv/linux/bits/in.h (IP_PMTUDISC_OMIT): Likewise.
            (IPV6_PMTUDISC_INTERFACE): Likewise.
            (IPV6_PMTUDISC_OMIT): Likewise.

Only the sysdeps/unix/sysv/linux/bits/in.h part is backported in this
patch.  The sysdeps/unix/sysv/linux/bits/fcntl-linux.h part was
completely superseded by glibc-rh1476120.patch, which uses
<linux/falloc.h> in favor of duplicated constants.

diff --git a/sysdeps/unix/sysv/linux/bits/in.h b/sysdeps/unix/sysv/linux/bits/in.h
index ac07d17d0e9dc9c9..23046d3b5a6b0d91 100644
--- a/sysdeps/unix/sysv/linux/bits/in.h
+++ b/sysdeps/unix/sysv/linux/bits/in.h
@@ -109,6 +109,8 @@
    Also incoming ICMP frag_needed notifications will be ignored on
    this socket to prevent accepting spoofed ones.  */
 #define IP_PMTUDISC_INTERFACE           4
+/* Like IP_PMTUDISC_INTERFACE but allow packets to be fragmented.  */
+#define IP_PMTUDISC_OMIT		5
 
 #define IP_MULTICAST_IF			32
 #define IP_MULTICAST_TTL 		33
@@ -224,6 +226,8 @@ struct in_pktinfo
 #define IPV6_PMTUDISC_WANT	1	/* Use per route hints.  */
 #define IPV6_PMTUDISC_DO	2	/* Always DF.  */
 #define IPV6_PMTUDISC_PROBE	3	/* Ignore dst pmtu.  */
+#define IPV6_PMTUDISC_INTERFACE	4	/* See IP_PMTUDISC_INTERFACE.  */
+#define IPV6_PMTUDISC_OMIT	5	/* See IP_PMTUDISC_OMIT.  */
 
 /* Socket level values for IPv6.  */
 #define SOL_IPV6        41
